File: 2-year/Java/Lab1/src/main/java/com/Starygin/Presentation/Scenarios/LoginScenario.java
package com.Starygin.Presentation.Scenarios;

import com.Starygin.Application.Contracts.UserService;
import com.Starygin.Presentation.Scenario;

import java.io.IOException;
import java.util.Scanner;

public class LoginScenario implements Scenario {
    private final UserService userService;

    public LoginScenario(UserService userService) {
        this.userService = userService;
    }

    @Override
    public String getName() {
        return "Login Scenario";
    }

    @Override
    public void run() {
        Scanner s = new Scanner(System.in);
        System.out.print("Enter your username: ");

        String userName = s.nextLine();

        System.out.print("Enter your password: ");

        String password = s.nextLine();

        try {
            userService.login(userName, password);
            System.out.println("Successfully logged in!");

        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: 2-year/OOP-CSharp/src/Lab5/Presentation/Lab5.Presentation.Console/Scenarios/Login/LoginScenario.cs
using Lab5.Application.Contracts.Users;
using Lab5.Application.Contracts.Users.OperationsResults;
using Spectre.Console;

namespace Lab5.Presentation.Console.Scenarios.Login;

public class LoginScenario : IScenario
{
    private readonly IUserService _userService;

    public LoginScenario(IUserService userService)
    {
        _userService = userService;
    }

    public string Name => "Login";

    public void Run()
    {
        string username = AnsiConsole.Ask<string>("Enter your username: ");
        string password = AnsiConsole.Prompt(
            new TextPrompt<string>("Enter your password: ").Secret());

        LoginResult result = _userService.Login(username, password);

        string message = result switch
        {
            LoginResult.Success => "Successful login",
            LoginResult.NotFound => "User not found",
            LoginResult.IncorrectPassword => "Incorrect password",
            _ => throw new ArgumentOutOfRangeException(nameof(result)),
        };

        AnsiConsole.WriteLine(message);
        AnsiConsole.Ask<string>("Ok");
    }
}